### PR TITLE
fix(router-lite): lifecycle hooks were invoked twice

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

## 📖 Description

This PR fixes the issue of the lifecycle hooks being called twice when registered as dependencies of custom elements.
Previously, the component dependencies were registered to the container when creating route context. This has caused the issue of same hooks being called twice because the hooks were registered twice (the one from the `Controller.$el` should happen normally). This PR removes the 2nd registration of dependencies.

### CODE REVIEW

File: packages/router-lite/src/route-context.ts [hunk 1]
```diff
@@ -166,7 +166,6 @@ export class RouteContext {
     );
 
     container.register(definition);
-    container.register(...component.dependencies);
```
Comment (bigopon) [inline packages/router-lite/src/route-context.ts:169]: 👍

File: packages/kernel/src/di.container.ts [hunk 2]
```diff
@@ -375,7 +375,7 @@ export class Container implements IContainer {
 
   public find<TType extends ResourceType, TDef extends ResourceDefinition>(kind: IResourceKind<TType, TDef>, name: string): TDef | null {
     const key = kind.keyFrom(name);
-    let resolver = this.res[key];
+    let  resolver = this.res[key];
```
Comment (bigopon) [inline packages/kernel/src/di.container.ts:378]: quite random addition :)

File: packages/router-lite/src/router.ts [hunk 3]
```diff
@@ -330,6 +330,7 @@ export class Router {
     return this._isNavigating;
   }
 
+  /** @internal */
```
Comment (bigopon) [inline packages/router-lite/src/router.ts:333]: 👍

File: packages/kernel/src/di.container.ts (hunk 2 above)
Comment (Sayan751) [reply packages/kernel/src/di.container.ts:378]: Lol! Sorry. I changed that file earlier, and later decided against that reverted that changes. Missed it somehow.

**Pull request merged**